docs/source/deploy.rst

```diff
@@ -0,0 +1,135 @@
+======
+Deploy
+======
+
+The deployment docs are split up in two parts. First there is the
+general section, which should always be followed. Then there is a cloud
+service provider specific section of which one provide should be chosen.
+
+Setup an example project
+========================
+
+1. Create a project directory of notebooks you wish to display. For this
+   tutorial we will clone voila and treat the notebooks folder as our
+   project root.
+
+   .. code:: bash
+
+       $ git clone git@github.com:QuantStack/voila.git
+       $ cd voila/notebooks/
+
+2. Add a requirements.txt file to the project directory. For this
+   tutorial we will copy the contents of the environment.yml of voila.
+   We omit xleaflet and xeus-cling because these require extra work that is
+   beyond the scope of this guide.
+
+   .. code:: txt
+
+       ipyvolume
+       bqplot
+       scipy
+       ipympl
+       voila==0.1.2
+       ipympl
+
+Cloud Service Providers
+=======================
+
+Deployment on Heroku
+--------------------
+
+Heroku.com is an attractive option if you want to try out deployment for
+free. You have limited computing hours, however the app will also
+automatically shutdown if it is idle.
+
+The general steps for deployment at Heroku can be found
+`here <https://devcenter.heroku.com/articles/getting-started-with-python>`__.
+High level instructions, specific to voila can be found below:
+
+1. Follow the steps of the official documentation to install the heroku
+   cli and login on your machine.
+2. Add a file named runtime.txt to the project directory with the following
+   content:
+
+   .. code:: txt
+
+       python-3.7.3
+
+3. Add a file named Procfile to the project directory with the
+   following content if you want to show all notebooks:
+
+   ::
+
+       web: voila —-port=$PORT --no-browser
+
+   Or the following if you only want to show one notebook:
+
+   ::
+
+       web: voila —-port=$PORT —-no-browser your_notebook.ipynb
+
+4. Initialize your git repo and commit your code
+
+   .. code:: bash
+
+       $ git init .
+       $ git add .
```

Comment:
I'm not sure we should encourage users to do `git add .`. They'll most probably be adding `ipynb_checkpoints` directories etc. to the index.

Would it make sense to have a placeholder that forces users to think about the files they want to add?

Reply:
resolved with placeholders in code, and text on what should at least be included